src/database/DataTypes/TransactionItem.js

```diff
@@ -70,12 +70,16 @@ export class TransactionItem extends Realm.Object {
    * N.B. For customer invoices, will create and delete transaction batches as appropriate.
    * N.B. Supplier invoices do not take effect on the rest of the stock until they
    * are finalised, whereas customer invoices immediately influence stock levels.
-   * @param {double} quantity The total quantity to set across all batches
+   * @param {Database} Realm DB wrapper
+   * @param {double} requestedQuantity try to set this quanity against all batches
+   * @param {double} referenceQuantity max available quantity or -1 to use requestedQuantity
    */
-  setTotalQuantity(database, quantity) {
+  setTotalQuantity(database, requestedQuantity, referenceQuantity = -1) {
```

Comment:
Actually `referenceQuantity` is always being passed in as `transactionItem.availableQuantity`, and come to think of it we can probably always just cap `requestedQuantity` at `availableQuantity` if the transaction `isOutgoing`